tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/class_conflicts_test.dart

```diff
@@ -44,6 +45,42 @@ void main() {
     );
   });
 
+  test(
+      'Given a model and a custom class with the same class name, then an error is collected that there is a collision in the class names.',
```

Comment:
Comment(Does not require fix): The description here is missing the `when` part. This is reflected by all tests in these files. So if it is fix is done, it should be reflected in all files.